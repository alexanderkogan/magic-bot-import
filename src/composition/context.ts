import * as React from 'react'
import { BookApi, BookApiImpl } from '../test-page/api/fetch-book-info'

export interface AppContext {
    books: BookApi
}
export interface AppContextProps {
    context: AppContext
}
export const productionContext: AppContext = {
    books: new BookApiImpl(),
}
export const AppContext: React.Context<AppContext> = React.createContext(productionContext)
