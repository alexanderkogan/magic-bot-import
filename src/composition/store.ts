import * as redux from 'redux'

export interface RootState {}

export function createStore(): redux.Store<RootState> {
    const reducer = () => ({})
    return redux.createStore(reducer)
}
