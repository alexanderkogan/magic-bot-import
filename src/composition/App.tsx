import * as React from 'react'
import { hot } from 'react-hot-loader/root'

import { AppContext } from './context'

import TestPage from '../test-page/TestPage'

import { BrowserRouter as Router, Switch, Route } from 'react-router-dom'
import Tab from '../pattern-library/Nav'
import TabItem from '../pattern-library/NavItem'

export interface AppProps {
    context: AppContext
}

export const App = (props: AppProps) => (
    <AppContext.Provider value={props.context}>
        <Router>
            <div>
                <Tab>
                    <TabItem to="/">Home</TabItem>
                    <TabItem to="/test">Pattern Library Test</TabItem>
                </Tab>

                <Switch>
                    <Route path="/test">
                        <TestPage />
                    </Route>
                    <Route path="/">lorem ipsum</Route>
                </Switch>
            </div>
        </Router>
    </AppContext.Provider>
)

export default hot(App)
