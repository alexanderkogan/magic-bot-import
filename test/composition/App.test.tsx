import * as React from 'react'

import { App } from '../../src/composition/App'
import { shallow } from '../test-utils'

describe('App', () => {
    it('renders shallow', () => {
        const element = shallow(<App />)
        expect(element).toMatchSnapshot()
    })
})
