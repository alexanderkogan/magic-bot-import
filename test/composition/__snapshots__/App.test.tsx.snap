// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders shallow 1`] = `<div />`;
